### PR TITLE
Rerender marker on child props change

### PR DESCRIPTION
**Pull request by KhaledSakr**

### CODE REVIEW

File: src/Marker.tsx [hunk 1]
```diff
@@ -38,6 +40,17 @@ export class Marker extends React.Component<MarkerProps, object> {
...
+    // Rerender the marker if child props change
+    if (React.Children.count(nextProps.children) > 0
+      && React.Children.count(this.props.children) > 0
+      && !isEqual(nextProps.children.props, this.props.children.props)) {
+      const { markersGroup } = this.context;
+      if (this.marker) {
+        markersGroup.removeObject(this.marker);
+      }
+      this.marker = this.renderChildren(nextProps.children, nextProps.lat, nextProps.lng)  
+      return
```
Comment (antoineraoulimpargo) [inline src/Marker.tsx]: The return is meaningless here please remove it,
There is no reason to stop code execution after that point as we might get out of sync between props and state.

File: src/Marker.tsx [hunk 2]
```diff
@@ -14,6 +15,7 @@ export interface MarkerProps extends H.map.Marker.Options, H.geo.IPoint {
   bitmap?: string;
   data?: any;
   draggable?: boolean;
+  children?: any;
```
Comment (antoineraoulimpargo) [inline src/Marker.tsx]: Just a really minor thing please use correct type here

Comment (KhaledSakr) [reply src/Marker.tsx]: I added the type React.ReactElement<any>

File: src/Marker.tsx [hunk 3]
```diff
@@ -14,6 +15,7 @@ export interface MarkerProps extends H.map.Marker.Options, H.geo.IPoint {
   bitmap?: string;
   data?: any;
   draggable?: boolean;
+  children?: React.ReactElement<any>;
```
Comment (antoineraoulimpargo) [inline src/Marker.tsx]: There is a bug report regarding ReactElement<any> https://github.com/DefinitelyTyped/DefinitelyTyped/issues/27805
Please use a generic node instead as we might run into a problem later on.
It's not important anyways as type safety is removed in prod.

**Pull request merged**